### DISCUSSION TITLE
Use CELERY_TASK_ALWAYS_EAGER and CELERY_TASK_EAGER_PROPAGATES in test.py?

### DISCUSSION THREAD

**Discussion by AhmedHathout** [cookiecutter/cookiecutter-django#5193] [General]
When I used cookiecutter more than a year ago to start a new project, these 2 celery settings were not in any of the settings file (local, test, production, base). I can see that they are there now in [local](https://github.com/cookiecutter/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/local.py#L115). I wonder if there is a reason for not adding them to `test.py` too. I actually think that it is more important in `test.py` (and they should be used whether we are using docker or not). Please let me know if you think I am missing something

**Comment by foarsitter** [comment]
Asking why we did x and not to y is a question in the category of 'why didnt you wrote this template in PHP?'. 

If you want the reason behind something find the commit and PR behind it. 

Quote the PR/commit if it not satisfies your beliefs and argument why it doesnt. 

If you believe the setting should be added to the test environment you need to provide some arguments. Do not hesitate as the popular meme about the internet states: if you want the correct answer about a topic provide a wrong answer. People happily prove you wrong :)

**Comment by browniebroke** [comment]
They are set in local, but only if Docker isn't selected, which I think makes sense. It's much easier to get started when the tasks are executed eagerly, since people are unlikely to remember to run a separate worker process. With docker compose, we can start this automatically, but not without.

I don't see a reason why they aren't defined in `test.py` indeed. I usually set them during tests on the projects I work on. PR welcome.
